### PR TITLE
gofmt: add page

### LINKED ISSUE #1607: page request: gofmt
webpage and cli documentation here - https://golang.org/cmd/gofmt/

Please go through the [CONTRIBUTING](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) page to understand what is the format we use for our pages.

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines 

Closes #1607

### CODE REVIEW

File: pages/common/gofmt.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# gofmt
+
+> Gofmt formats Go programs. It uses tabs for indentation and blanks for alignment.
```
Comment (agnivade) [inline pages/common/gofmt.md]: How about `Tool for formatting Go source code` ? It matches with the `go` page. We can remove the next sentence. The exact details of the formatting specification is not required here.

File: pages/common/gofmt.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+- Format a file and print it to output:
```
Comment (agnivade) [inline pages/common/gofmt.md]: Probably `Format a file and display the result to console:` is clearer.

File: pages/common/gofmt.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+`gofmt {{source.go}}`
+
+- Format a file and rewrite original:
```
Comment (agnivade) [inline pages/common/gofmt.md]: `Format a file, overwriting the original file in-place.`

File: pages/common/gofmt.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+`gofmt {{source.go}} -w`
+
+- Print all (including spurious) errors:
+
+`gofmt {{source.go}} -e`
+
+- Apply custom rule as string "pattern -> replacement":
```
Comment (agnivade) [inline pages/common/gofmt.md]: We can remove this example. It is not used very often.

File: pages/common/gofmt.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+- Format a file and print it to output:
+
+`gofmt {{source.go}}`
+
+- Format a file and rewrite original:
+
+`gofmt {{source.go}} -w`
+
+- Print all (including spurious) errors:
```
Comment (agnivade) [inline pages/common/gofmt.md]: Please add a `gofmt -s -w {{source.go}}` example above this. That is the most commonly used convention.

**Comment by nalinbhardwaj** [comment]
@agnivade done.

**Comment by agnivade** [comment]
Thanks !

**Pull request merged**